rs/replicated_state/src/canister_state/system_state.rs

```diff
@@ -1073,7 +1075,44 @@ impl SystemState {
 
     /// Extracts the next inter-canister or ingress message (round-robin).
     pub(crate) fn pop_input(&mut self) -> Option<CanisterMessage> {
-        self.queues.pop_input()
+        Some(match self.queues.pop_input()? {
+            CanisterInput::Ingress(msg) => CanisterMessage::Ingress(msg),
+            CanisterInput::Request(msg) => CanisterMessage::Request(msg),
+            CanisterInput::Response(msg) => CanisterMessage::Response(msg),
+            CanisterInput::DeadlineExpired(callback_id) => {
+                self.to_reject_response(callback_id, "Call deadline has expired.")
+            }
+            CanisterInput::ResponseDropped(callback_id) => {
+                self.to_reject_response(callback_id, "Response was dropped.")
+            }
+        })
+    }
+
+    /// Generates a reject response for the given callback ID with the given
+    /// message.
+    ///
+    /// Panics if the `CallContextManager` does not hold a callback with the given
+    /// `CallbackId`.
+    fn to_reject_response(&self, callback_id: CallbackId, message: &str) -> CanisterMessage {
+        let call_context_manager = self.call_context_manager().unwrap();
+        let callback = call_context_manager.callbacks().get(&callback_id).unwrap();
+
+        CanisterMessage::Response(
+            Response {
+                originator: callback.originator,
+                respondent: callback.respondent,
+                originator_reply_callback: callback_id,
+                refund: Cycles::zero(),
+                response_payload: Payload::Reject(RejectContext::new_with_message_length_limit(
+                    // TODO(MR-552): Use SysUnknown instead.
+                    RejectCode::SysTransient,
```

Comment:
This should meanwhile be merged, right?

Reply:
Replaced.

Although technically we should have waited for the `SysUnknown` addition to doll back to all subnets, we're not producing best-effort messages and thus this code path is unreachable on mainnet.

Reply:
Makes sense. On top of that the next RC is very likely gonna contain the `SysUnknown` change whereas it is very likely not gonna contain the change here. So changing it here will even likely end up in two stages on mainnet.